src/ui/view/governance/details/details.tsx

```diff
@@ -51,19 +53,48 @@ const Paragraph: FC<ParagraphProps> = ({ paragraph, theme }) => {
     }
   }, [])
 
+  const DIDDescription = useMemo(
+    () =>
+      paragraph.description.length > 22
+        ? [paragraph.description.slice(0, 16), '...', paragraph.description.slice(-4)].join('')
+        : paragraph.description,
+    [paragraph.description]
+  )
+
+  const isDID = useMemo(
+    () =>
+      (description: string): boolean =>
+        description.startsWith('did:key:'),
+    []
+  )
```

Comment:
Thanks! You're right I this could be improved. Looking at the "did" syntax @ccamel linked in that comment I've modified the regex a bit so it captures less false positives, seeing as only some special characters are permitted: `/^did:[a-zA-Z0-9]+:[a-zA-Z0-9.\-_%]+$/`

Reply:
Great! Your proposal is more in line with the W3C specification for DIDs